pkg/pillar/scripts/device-steps.sh

```diff
@@ -58,6 +61,21 @@ wait_for_touch() {
     fi
 }
 
+#waits for a file ($1) for a given time ($2)
+wait_for_file() {
```

Comment:
wait_for_done $agentname $time?

Reply:
See my question above, but can't this be done by, effectively, introducing a command line utility to talk on our pubsub?

Reply:
We can have invoke a 10-40 Mbyte go binary to wait for something on pubsub. A bit simpler than the waitforaddr.go we have to wait for nim to get IP addresses. We can make such a waitforagentdone smaller since it doesn't need http etc, If we do we can replace the wait for the agent starting in the script (which looks for the .touch file) as well.

Reply:
As I said above, I think this should be called wait_for_done $agentname $time

Reply:
Added `wait_for_done()`.
Had to retain `wait_for_file()` to check on `/config/uuid`.

Reply:
Can't you instead wait for client.done?

Reply:
Hey @eriknordmark -- I think in general I'd prefer wrapping all this into a pubsub -- as you're saying we can make sure it is a pretty small standalone CLI that would allow anything to walk on the pubsub.

That said, I'd rather see this PR merged as-is and we can quickly follow up with pubsub refactoring.

Does this make sense to you @eriknordmark and @adarsh-zededa ?